### DISCUSSION TITLE
How to avoid indexing a production deployment that does not have a connected custom domain?

### DISCUSSION THREAD

**Discussion by cbdeveloper** [vercel/community#1060] [Help] [answered]
I'm migrating a project and I don't want the `production` deployment to be indexed by robots until I transfer and connect my custom domain it. What would be the best way to achieve that?

**Comment by amyegan** [comment, marked as answer]
Hi @cbdeveloper! You can use a `x-robots-tag: noindex` header to prevent the site from being indexed. That's what Vercel already does automatically for Preview deployments to prevent negative SEO impact from duplicate content. There's a guide called ["Are Vercel Preview Deployments indexed by search engines?"](https://vercel.com/guides/are-vercel-preview-deployment-indexed-by-search-engines) which covers the topic in more detail. That should give you the information that you need to implement the solution for production deployment!

**Reply by khanakia** [reply]
Vercel is not adding the `x-robots-tag: noindex` you can verify `curl -I https://hustler-icu-site.vercel.app/` then why docs say it's automatically added?

**Reply by amyegan** [reply]
It looks like `hustler-icu-site.vercel.app` is associated with a Production deployment, so it wouldn't have `x-robots-tag: noindex` by default. If want to avoid indexing a production deployment, you need to add the header using the method recommended by your site's framework or with the [`header` object](https://vercel.com/docs/project-configuration#project-configuration/headers) in a `vercel.json` file.

**Comment by xdajay** [comment]
What a mess is this.
Vercel should by default block google accessing any of preview urls  even for production url still under .vercel.app
